dashboard/src/components/Header/NamespaceSelector.tsx

```diff
@@ -44,11 +58,23 @@ class NamespaceSelector extends React.Component<INamespaceSelectorProps> {
           promptTextCreator={this.promptTextCreator}
           clearable={false}
         />
+        <NewNamespace
+          modalIsOpen={this.state.modalIsOpen}
+          namespace={this.state.newNamespace}
+          onConfirm={this.onConfirmNewNS}
+          onChange={this.handleChangeNewNS}
+          closeModal={this.closeModal}
+          error={error && error.action === "create" ? error.error : undefined}
+        />
       </div>
     );
   }
 
   private handleNamespaceChange = (value: any) => {
+    if (value.value === "new") {
+      this.setState({ modalIsOpen: true });
+      return;
+    }
```

Comment:
Out of interest, what happens if I create a new namespace named "new", and then try to select it? :P (depending on the answer, you might need a token which is not a valid namespace).

Reply:
_good catch!_
I always wondered why the All Namespaces choice used `_all` (it does because `_all` is an invalid namespace  name).